docs/cudf/source/developer_guide/cudf_pandas.md

```diff
@@ -0,0 +1,119 @@
+Wrapped objects and proxy objects are instances of wrapped types and proxy types, respectively.
+In the snippet below `s1` and `s2` are wrapped objects and `s3` is a fast-slow proxy object.
+Also note that the module `xpd` is a wrapped module and contains cuDF and Pandas modules as attributes.
+  ```python
+  import cudf.pandas
+  cudf.pandas.install()
+  import pandas as xpd
+
+  cudf = xpd._fsproxy_fast
+  pd = xpd._fsproxy_slow
+
+  s1 = cudf.Series([1,2])
+  s2 = pd.Series([1,2])
+  s3 = xpd.Series([1,2])
+  ```
+
+Note that users should never have to interact with the wrapped objects directly in this way.
+This code is purely for demonstrative purposes.
+
+#### The Different Kinds of Proxy Types:
+In `cudf.pandas`, there are two main kinds of proxy types: final types and intermediate types.
+
+##### Final and Intermediate Proxy Types:
+Final types are types for which known operations exist for converting an object of a "fast" type to a "slow" type and vice versa.
+For example, `cudf.DataFrame` can be converted to Pandas using the method `to_pandas`, and `pd.DataFrame` can be converted to cuDF using the function `cudf.from_pandas`.
+Intermediate types are the types of the results of operations invoked on final types.
+For example, `xpd.DataFrameGroupBy` is an intermediate type that will be created during a groupby operation on the final type `xpd.DataFrame`.
+
+##### Attributes and Callable Proxy Types:
+Final proxy types are typically classes or modules, both of which have attributes.
+Classes also have methods.
+These attributes and methods must be wrapped as well to support the fast-slow proxy scheme.
+
+#### Creating New Proxy Types:
+`_FinalProxy` and `_IntermediateProxy` types are created using the functions `make_final_proxy_type` and `make_intermediate_proxy` type, respectively. Creating a new final type looks like this.
+
+```python
+DataFrame = make_final_proxy_type(
+    "DataFrame",
+    cudf.DataFrame,
+    pd.DataFrame,
+    fast_to_slow=lambda fast: fast.to_pandas(),
+    slow_to_fast=cudf.from_pandas,
+)
+```
+
+### The Fallback Mechanism
+The `_fast_slow_function_call` is the all-important function that we use to call operations the fast way (using cuDF) and if that fails, the slow way (using Pandas).
+The is also known as the fallback mechanism.
+The function looks like this:
+```python
+def _fast_slow_function_call(func: Callable, *args, **kwargs):
+    try:
+        ...
+        fast_args, fast_kwargs = _fast_arg(args), _fast_arg(kwargs)
+        result = func(*fast_args, **fast_kwargs)
+        ...
+    except Exception:
+        ...
+        slow_args, slow_kwargs = _slow_arg(args), _slow_arg(kwargs)
+        result = func(*slow_args, **slow_kwargs)
+        ...
+    return _maybe_wrap_result(result, func, *args, **kwargs), fast
+```
+As you can see the function attempts to call `func` the fast way using cuDF and if any `Exception` occurs, it calls the function using Pandas.
```

Comment:
```suggestion
As we can see the function attempts to call `func` the fast way using cuDF and if any `Exception` occurs, it calls the function using Pandas.
```

Although I know I might be the minority in preferring the active "we" form in documentation.